linalg/hypre.cpp

```diff
@@ -1549,10 +1561,19 @@ HypreParMatrix *HypreParMatrix::ExtractSubmatrix(const Array<int> &indices,
                               CF_marker, NULL, &cpts_global);
 
    // Extract submatrix into *submat
+#ifdef hypre_IntArrayData
+   hypre_ParCSRMatrixExtractSubmatrixFC(A, hypre_IntArrayData(CF_marker),
+                                        cpts_global, "FF", &submat,
+                                        threshold);
```

Comment:
Is the issue with the latest HYPRE `master` branch that now `hypre_ParCSRMatrixExtractSubmatrixFC` expects the second parameter to be on device? If so, we don't need to use the new `hypre_IntArray` -- we can simply ask the `mfem::Array<HYPRE_Int>`, `CF_marker` to move the data to device and give us the device pointer.

Reply:
Hi @v-dobrev, `hypre_ParCSRMatrixExtractSubmatrixFC` expects the second parameter to be on the host. Currently that function works only for host. @waynemitchell, perhaps there's a plan to make it work for device also? 

The issue with the latest HYPRE `master` branch is that it expects the second parameter of `hypre_ParCSRMatrixExtractSubmatrixFC` to be of type `hypre_IntArray` instead of `HYPRE_Int *`

Reply:
Wait, this is the prototype in the latest hypre master: https://github.com/hypre-space/hypre/blob/c9b1679970335a770978dbc36d83b6394ecc29b8/src/parcsr_mv/_hypre_parcsr_mv.h#L837 and the second argument is `HYPRE_Int *`. What am I missing?

Reply:
@v-dobrev @victorapm There are still many features in hypre that have not been ported to the device, and we rely on unified memory in order to ensure these features work when building with cuda. `hypre_ParCSRMatrixExtractSubmatrixFC`  does not have a GPU implementation, so it needs the CF marker to be a unified memory pointer in order to work. The parameter type here should still just be HYPRE_Int *.